docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,214 @@
+User Customization & Persistence
+================================
+
+Context
+-------
+
+.. todo::
+
+   This section will be handled by the requirements PR.
+
+
+Design Choices
+--------------
```

Comment:
This design only discusses the choice of storage (method and media). What about the concerns of translating these user configurations into services configurations? Are the services supposed to know about our user configs and parse them? Are we (Salt) deriving actual service configs from our user configs? Why Salt and not a K8s Operator?

Reply:
I added a `How it works section` that takes care of the questions raised here. This might need validation from You